rudder-lang/src/parser.rs

```diff
@@ -190,25 +191,47 @@ macro_rules! sequence {
     ( { $($f:ident : $parser:expr;)* } => $output:expr ) => {
         move |i| {
             $(
-                let (i,$f) = $parser (i)?;
+                // save last line (cursor is updated at parser calls). context of an potential error
+                let focalized_error = get_focalized_input(i);
+                // intercept error to update its context if it should lead to a handled compilation error
+                let (i, $f) = match $parser (i) {
+                    Ok(res) => res,
+                    Err(e) => return Err(update_error_context(e, focalized_error))
+                };
             )*
             Ok((i, $output))
         }
     };
 }
+
 /// wsequence is the same a sequence, but we automatically insert space parsing between each call
 macro_rules! wsequence {
     ( { $($f:ident : $parser:expr;)* } => $output:expr ) => {
         move |i| {
             $(
-                let (i,$f) = $parser (i)?;
+                // save last line (cursor is updated at parser calls). context of an potential error
+                let focalized_error = get_focalized_input(i);
+                // intercept error to update its context if it should lead to a handled compilation error
+                let (i, $f) = match $parser (i) {
+                    Ok(res) => res,
+                    Err(e) => return Err(update_error_context(e, focalized_error))
+                };
                 let (i,_) = strip_spaces_and_comment(i)?;
             )*
             Ok((i, $output))
         }
     };
 }
 
+/// Tool function allowing to save last state of parsing offset line before any error
+fn get_focalized_input(i: PInput) -> PInput {
```

Comment:
Is focalized the right term, isn't there a more explicit word ?

Reply:
renamed